torchgeo/datasets/urban3d.py

```diff
@@ -0,0 +1,281 @@
+
+        mask = self._load_target(files["binary_mask"])
+        mask = mask.to(torch.long)  # type: ignore[attr-defined]
+
+        instances = self._load_image(files["instance_mask"]).squeeze(dim=0)
+        instances = instances.to(torch.long)  # type: ignore[attr-defined]
+
+        sample = {"image": image, "mask": mask, "instances": instances}
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def _load_image(self, path: str) -> Tensor:
+        """Load a single image.
+
+        Args:
+            path: path to image
+
+        Returns:
+            the image
+        """
+        with rasterio.open(path) as f:
+            array = f.read(out_dtype="float32")
+            tensor: Tensor = torch.from_numpy(array)  # type: ignore[attr-defined]
+            return tensor
+
+    def _load_target(self, path: str) -> Tensor:
+        """Load a single target.
+
+        Args:
+            path: path to target
+
+        Returns:
+            the target
+        """
+        mask = self._load_image(path).squeeze(dim=0)
+        mask[mask == 2] = 0  # 2 = background
+        mask[mask == 6] = 1  # 6 = building
+        mask[mask == 65] = 1  # 65 = building partially overlapping with nodata
+        return mask
+
+    def _load_files(self) -> List[Dict[str, str]]:
+        """Return the paths of the files in the dataset.
+
+        Returns:
+            list of dicts containing paths for each sample
+        """
+        image_root = os.path.join(self.root, self.directories[self.split], "Inputs")
+        target_root = os.path.join(self.root, self.directories[self.split], "GT")
+        basenames = [
+            os.path.basename(f) for f in glob.glob(os.path.join(image_root, "*.tif"))
+        ]
+        prefixes = set([os.path.splitext(f)[0].rsplit("_", 1)[0] for f in basenames])
+
+        files = []
+        for prefix in sorted(prefixes):
+            files.append(
+                dict(
+                    rgb=os.path.join(image_root, f"{prefix}_RGB.tif"),
+                    dtm=os.path.join(image_root, f"{prefix}_DTM.tif"),
+                    dsm=os.path.join(image_root, f"{prefix}_DSM.tif"),
+                    binary_mask=os.path.join(target_root, f"{prefix}_GTL.tif"),
+                    instance_mask=os.path.join(target_root, f"{prefix}_GTI.tif"),
+                )
+            )
+        return files
+
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset.
+
+        Raises:
+            RuntimeError: if the dataset is not found
+        """
+        # Check if the files already exist
+        exists = []
+        for directory in self.directories.values():
+            exists.append(os.path.exists(os.path.join(self.root, directory)))
+        if all(exists):
+            return
+
+        raise RuntimeError(
+            "Dataset not found in `root` directory, "
```

Comment:
We should be specifying which directory we try like in other datasets, this error message isn't useful as is. Yes, there are some datasets where we don't do this, but existing datasets with bad error messages aren't an excuse for new datasets with bad error messages.